packages/hydrogen/src/storefront.ts

```diff
@@ -368,15 +368,15 @@ export function createStorefrontClient<TI18n extends I18nBase>(
 
     const {data, errors} = body as GraphQLApiResponse<T>;
 
-    if (errors?.length) {
+    if (!data && errors?.length) {
```

Comment:
Are we sure that during a full API failure, data is always undefined/null? Or does it ever show up as an empty object or something?

Reply:
Not sure if I understood it correctly but: should we just `console.error(Error)` the errors that we are currently swallowing and keep throwing the other ones like we do now?
That's non-breaking and at least gives the user more information, but I'm not sure if it's enough. I.e. is this kind of api error important enough so that the developer needs to fork logic? Because in that case we should also throw them somehow (or return them as you proposed originally).